test/config.sh

```diff
@@ -2,6 +2,8 @@
 
 set -e
 
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
```

Comment:
This is horrific. As I understand it the complication arises from the fact that we are 'sourcing' the script. @dpw is there a better way.